autokey-blueprints/1-storage-bucket/variables.tf

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+variable "suffix" {
+  description = "A suffix to be used as an identifier for resources. (e.g., suffix for KMS Key, Keyring, SAs, etc.)."
+  type        = string
+  default     = ""
+}
+
+variable "autokey_folder_id" {
+  description = "Autokey folder ID to be created or used."
+  type        = string
+}
+
+variable "autokey_key_project_id" {
+  description = "GCP project ID to be used for KMS Autokey keys."
+  type        = string
+}
+
+variable "autokey_handle_name" {
+  description = "Autokey Ken Handle name."
```

Comment:
According to [KMS API docs](https://cloud.google.com/kms/docs/reference/rest/v1/projects.locations.keyHandles/create?rep_location=global): "if not provided by the caller, a new UUID is used."

If we explicit pass a name, it will be used. We have [this assert](https://github.com/terraform-google-modules/terraform-google-kms/blob/8a28c2fcf5798b5b2f02a7ea31098277e3c9e04e/test/integration/autokey_example/autokey_example_test.go#L79) on autokey's module validating the "name" attribute.

Reply:
The short answer is yes, the name is being honoured. But here's why:

According to [KMS API docs](https://cloud.google.com/kms/docs/reference/rest/v1/projects.locations.keyHandles/create?rep_location=global) for `keyHandleId`: "_if not provided by the caller, a new UUID is used._". But `name` it is actually required when using [kms_key_handle](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/kms_key_handle) resource and resource's `name` is mapped to API's `keyHandleId`.

I enabled the terraform DEBUG mode to validate if the `name` is being honored:

See this example of a Key Handle POST creation by terraform:
![image](https://github.com/user-attachments/assets/82a69b47-b1b2-4f8d-9d4e-98cdd86eb081)
- Notice that, the `keyHandleId` passed to the API is the according to the `name` that the resource received;
- The response returned a UUID related to the operation. It's not the Key Handle ID itself.

When terraform sends a GET for this resource, using the `name` passed on creation as the Key Handle ID, the expected object is returned.
![image](https://github.com/user-attachments/assets/fbd05fde-03a0-4970-b383-556fecdd05cd)